src/main/java/org/apache/parquet/hadoop/BdecParquetWriter.java

```diff
@@ -283,7 +282,7 @@ public void prepareForWrite(RecordConsumer recordConsumer) {
 
     @Override
     public void write(List<Object> values) {
-      List<ColumnDescriptor> cols = schema.getColumns();
+      List<Type> cols = schema.getFields();
```

Comment:
What's the difference between schema.getColumns and getFields?
Also just to confirm getFields is supposed to return top level columns, and not any subcolumns?

Reply:
pl add comment

Reply:
Added comment.